pyartcd/pyartcd/pipelines/scheduled/schedule_ocp4_scan.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+
+import click
+
+from pyartcd import util, jenkins
+from pyartcd.cli import cli, click_coroutine, pass_runtime
+from pyartcd.locks import Lock, LockManager
+from pyartcd.runtime import Runtime
+
+
+async def run_for(version: str, runtime: Runtime):
+    # Skip if locked
+    lock = Lock.BUILD
+    lock_name = lock.value.format(version=version)
+    lock_manager = LockManager.from_lock(lock)
+    try:
+        locked = await lock_manager.is_locked(lock_name)
+    finally:
+        await lock_manager.destroy()
+    if locked:
+        runtime.logger.info('[%s] Locked, skipping', version)
+        return
+
+    # Skip if frozen
+    if not await util.is_build_permitted(version, doozer_working=str(runtime.working_dir / "doozer_working-" / version)):
+        runtime.logger.info('[%s] Not permitted, skipping', version)
+        return
+
+    # Schedule scan
+    runtime.logger.info('[%s] Scheduling ocp4-scan', version)
+    jenkins.start_ocp4_scan(version=version)
```

Comment:
There's a parent timeout of 60 mins https://github.com/openshift-eng/aos-cd-jobs/pull/4226/files . I'm thinking let's include that in the ocp4_scan job itself?

Reply:
:+1:   Done [here](https://github.com/openshift-eng/aos-cd-jobs/pull/4226/commits/e38d0b10b6e2d3443d74731ecb8f8c675266e4da)